DataOnion/Auth/SlidingExpirationStrategy.cs

```diff
@@ -0,0 +1,108 @@
+using StackExchange.Redis;
+
+namespace DataOnion.Auth
+{
+    public class SlidingExpirationStrategy<T> : IAuthServiceStrategy<T>
+        where T : class, IAuthStorable<T>
+    {
+        public static string ExpiryKey { get; } = "expiration";
+
+        private readonly IDatabase _database;
+        private readonly TimeSpan _timeout;
+        private readonly TimeSpan? _absoluteExpiration;
+        private readonly string _keyPrefix;
+        private readonly Func<HashEntry[], T> _makeFromHash;
+
+        private string _expirationTimeStr => DateTimeOffset.UtcNow
+            .Add(_timeout)
+            .ToUnixTimeSeconds()
+            .ToString();
+
+        public SlidingExpirationStrategy(
+            IDatabase database,
+            TimeSpan timeout,
+            TimeSpan? absoluteExpiration,
+            string keyPrefix,
+            Func<HashEntry[], T> makeFromHash
```

Comment:
we should wrap these parameters in their own type so we can more reliably retrieve them from DI and reduce a chance for unintended collisions